icu4c/source/i18n/coll.cpp

```diff
@@ -167,7 +167,7 @@ class ICUCollatorService : public ICULocaleService {
     }
     
     virtual UObject* handleDefault(const ICUServiceKey& key, UnicodeString* actualID, UErrorCode& status) const override {
-        LocaleKey& lkey = (LocaleKey&)key;
+        LocaleKey& lkey = dynamic_cast<LocaleKey&>(const_cast<ICUServiceKey&>(key));
```

Comment:
If I change this line to static_cast<> I got the followng runtime error during test under CFI
```
TestGetLocale {
coll.cpp:170:27: runtime error: control flow integrity check for type 'icu::LocaleKey' failed during base-to-derived cast (vtable address 0x7f93915fa730)
0x7f93915fa730: note: invalid vtable
 93 7f 00 00  a0 7e 4c 91 93 7f 00 00  80 7b 4c 91 93 7f 00 00  a0 7b 4c 91 93 7f 00 00  f0 59 4c 91
              ^ 
coll.cpp:170:27: note: check failed in ../../lib/libicui18n.so.73, vtable located in ../../lib/libicuuc.so.73
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior coll.cpp:170:27 in 
```